Realm/Tests/KVOTests.mm

```diff
@@ -506,14 +506,14 @@ - (void)testRemoveObserver {
     XCTAssertNoThrow([obj addObserver:self forKeyPath:@"int32Col" options:0 context:context1]);
     XCTAssertThrows([obj removeObserver:self forKeyPath:@"int32Col" context:context2]);
     XCTAssertNoThrow([obj removeObserver:self forKeyPath:@"int32Col" context:context1]);
-
+    
```

Comment:
delete the whitespace

Reply:
@tgoyne I guess you are right, this function is overly optimistic about what to expect, this works every time the query is structured in a way that the subscripts are declared the expected way.  We do have to work with NSPredicate limitations and the only way I see we can declare subscripts in a field and not get an error from NSPredicate is this way (using subscripts)
If we for example use an NSPredicate with this structure `anyCol[0]['key']['all']`, NSPredicate will return a nested NSExpression where the first item is another NSExpression which nests as many NSExpressions we have as subscripts.
We are applying this paths to the query in case is a mixed value, we cannot know if the wrapped value is a collection or not from the property but only from the value at runtime, also even if we are querying against it, we cannot be sure the wrapped value is of that type. We are trusting core to do this verifications and not return any data if that's the case.

Reply:
Aso, fixed some hardcoded values in this function